spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-configmap-event-reload/src/test/java/org/springframework/cloud/kubernetes/client/configmap/event/reload/ConfigMapEventReloadIT.java

```diff
@@ -0,0 +1,369 @@
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.AppsV1Api;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.apis.NetworkingV1Api;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1ConfigMapBuilder;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1Ingress;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Service;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.k3s.K3sContainer;
+import reactor.netty.http.client.HttpClient;
+import reactor.util.retry.Retry;
+import reactor.util.retry.RetryBackoffSpec;
+
+import org.springframework.cloud.kubernetes.integration.tests.commons.Commons;
+import org.springframework.cloud.kubernetes.integration.tests.commons.K8SUtils;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.cloud.kubernetes.integration.tests.commons.Commons.processExecResult;
+import static org.springframework.cloud.kubernetes.integration.tests.commons.K8SUtils.createApiClient;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+/**
+ * @author wind57
+ */
+class ConfigMapEventReloadIT {
+
+	private static final String IMAGE_NAME = "spring-cloud-kubernetes-client-configmap-event-reload";
+
+	private static final String NAMESPACE = "default";
+
+	private static String deploymentName;
+
+	private static String serviceName;
+
+	private static String ingressName;
+
+	private static String leftConfigMapName;
+
+	private static String rightConfigMapName;
+
+	private static String rightWithLabelConfigMapName;
+
+	private static CoreV1Api api;
+
+	private static AppsV1Api appsApi;
+
+	private static NetworkingV1Api networkingApi;
+
+	private static K8SUtils k8SUtils;
+
+	private static final K3sContainer K3S = Commons.container();
+
+	@BeforeAll
+	static void beforeAll() throws Exception {
+		K3S.start();
+		Commons.validateImage(IMAGE_NAME, K3S);
+		Commons.loadSpringCloudKubernetesImage(IMAGE_NAME, K3S);
+		createApiClient(K3S.getKubeConfigYaml());
+		createNamespaces();
+		api = new CoreV1Api();
+		appsApi = new AppsV1Api();
+		networkingApi = new NetworkingV1Api();
+		k8SUtils = new K8SUtils(api, appsApi);
+
+		k8SUtils.setUpClusterWide(NAMESPACE, Set.of("left", "right"));
+	}
+
+	@AfterAll
+	static void afterAll() throws Exception {
+		deleteNamespaces();
+		Commons.cleanUp(IMAGE_NAME, K3S);
+	}
+
+	/**
```

Comment:
you can read the comments to understand what each test does and why. same tests are present in both clients. I decided to write integration tests only for confgmap functionality, mainly because _almost_ the same code is present for secrets. We already had integration tests for secrets informers anyway, but they are slightly simpler. Still, imo, this is enough to validate the logic.